src/AltinnCore/Designer/Startup.cs

```diff
@@ -28,212 +28,214 @@
+            services.AddSingleton<IViewRepository, ViewRepository>();
+            services.AddSingleton<IGitea, GiteaAPIWrapper>();
+            services.AddSingleton<ISourceControl, SourceControlSI>();
+            services.AddSingleton(Configuration);
+            services.TryAddSingleton<IHttpContextAccessor, HttpContextAccessor>();
+
+            services.AddResponseCompression();
+
+            // TODO: Figure out how appsettings.json parses values and merges with environment variables and use these here.
+            // Since ":" is not valid in environment variables names in kubernetes, we can't use current docker-compose environment variables
+            string repoLocation = null;
+
+            if (Environment.GetEnvironmentVariable("ServiceRepositorySettings__RepositoryLocation") != null)
             {
-              // Add Custom Event handler to be able to redirect users for authentication upgrade
-              OnRedirectToAccessDenied = NotAuthorizedHandler.RedirectToNotAuthorized
-            };
-          });
-
-      var mvc = services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_2_1);
-      mvc.Services.Configure<MvcOptions>(options =>
-      {
-        // Adding custom modelbinders
-        options.ModelBinderProviders.Insert(0, new AltinnCoreApiModelBinderProvider());
-        options.ModelBinderProviders.Insert(1, new AltinnCoreCollectionModelBinderProvider());
-      });
-      mvc.AddXmlSerializerFormatters();
-
-      services.Configure<RazorViewEngineOptions>(options =>
-      {
-        options.FileProviders.Add(
-                  new AltinnViewFileProvider(repoLocation + "{0}/{1}/{2}/Views/", repoLocation + "{0}/{1}/{2}/Packages/", repoLocation + "{0}/{1}/{2}/", loadFromPackage));
-      });
-
-      services.AddLocalization();
-      services.Configure<RequestLocalizationOptions>(
-          options =>
-          {
-            var supportedCultures = new List<CultureInfo>
-                  {
+                repoLocation = Environment.GetEnvironmentVariable("ServiceRepositorySettings__RepositoryLocation");
+            }
+            else
+            {
+                repoLocation = Configuration["ServiceRepositorySettings:RepositoryLocation"];
+            }
+
+
+            if (!Directory.Exists(repoLocation))
+            {
+                Directory.CreateDirectory(repoLocation);
+            }
+
+            services.Configure<ServiceRepositorySettings>(Configuration.GetSection("ServiceRepositorySettings"));
+            services.Configure<TestdataRepositorySettings>(Configuration.GetSection("TestdataRepositorySettings"));
+            services.Configure<GeneralSettings>(Configuration.GetSection("GeneralSettings"));
+
+            // Configure Authentication
+            // Use [Authorize] to require login on MVC Controller Actions
+            services.AddAuthentication(CookieAuthenticationDefaults.AuthenticationScheme)
```

Comment:
Since most of the file is indentation changes. This is where i changed to add options to the authentication.